src/esy/PackageEnvironment.js

```diff
@@ -288,6 +288,11 @@ function computeEnvVarsForPackage(
   packageInfo: PackageInfo
 ) {
   let {rootDirectory, packageJson, normalizedName} = packageInfo;
+  // Do not compute package environment for this one package, if it
+  // does not have an `esy` field.
+  if (packageJson.esy && packageJson.esy.__noEsyConfigPresent) {
```

Comment:
Specifically, a package may not export the `PATH` variable, yet since `esy` automatically handles the env var `PATH`, it won't show up in the `exportedEnv`.